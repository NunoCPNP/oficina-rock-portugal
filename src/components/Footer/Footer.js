import { AiFillFacebook, AiFillInstagram, AiFillPhone, AiOutlineWhatsApp, AiTwotoneMail } from 'react-icons/ai'
import Image from 'next/image'
import { SiTiktok } from 'react-icons/si'
import useTranslation from 'next-translate/useTranslation'

import Copyright from '../Copyright'

import { ContactsContainer, GridContainer, GridItem, MethodsContainer, Spacer, StyledFooter } from './Footer.styles'

const Footer = () => {
  const { t } = useTranslation('common')

  return (
    <StyledFooter data-testid="footer">
      <GridContainer>
        <GridItem>
          <div>
            <h4>{t(`who-we-are`)}</h4>
          </div>
          <div>
            <h4>{t(`privacy-policies`)}</h4>
          </div>
          <div>
            <h4>{t(`return-and-refund`)}</h4>
          </div>
          <div>
            <h4>{t(`faq`)}</h4>
          </div>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`contacts`)}</h3>
          <ContactsContainer>
            <div>
              <AiFillPhone data-testid="react-icon" />
              <AiOutlineWhatsApp data-testid="react-icon" />
              <span>+351 964 631 183</span>
            </div>
            <div>
              <AiFillFacebook data-testid="react-icon" />
              <AiFillInstagram data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
            <div>
              <AiTwotoneMail data-testid="react-icon" />
              <span>oficinarockpt@gmail.com</span>
            </div>
            <div>
              <SiTiktok data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
          </ContactsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`payment-methods`)}</h3>
          <MethodsContainer>
            <div>
              <Image src="/payment_methods.png" alt={t(`payment-methods`)} width={160} height={35} />
            </div>
          </MethodsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`newsletter`)}</h3>
          <Spacer />
        </GridItem>
      </GridContainer>
      <Copyright />
    </StyledFooter>
  )
}

export default Footer
