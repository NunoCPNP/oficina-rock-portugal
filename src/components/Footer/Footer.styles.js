import styled from '@emotion/styled'
import mq from '@/styles/mediaQueries'

export const StyledFooter = styled.footer`
  background-color: var(--ghostWhite);
`

export const GridContainer = styled.div`
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;

  ${mq('m')} {
    grid-template-columns: 1fr 1fr;
  }

  ${mq('s')} {
    grid-template-columns: 1fr;
  }
`

export const GridItem = styled.div`
  line-height: 3rem;
  h3 {
    text-transform: uppercase;
    padding-bottom: 1rem;
  }

  h4 {
    font-weight: 400;
  }
`

export const ContactsContainer = styled.div`
  padding-top: 1rem;

  div {
    display: flex;
    align-items: center;
    margin-bottom: 0.4rem;

    svg {
      width: 2.2rem;
      height: 2.2rem;
      margin-right: 0.6rem;
    }
  }
`

export const MethodsContainer = styled.div`
  padding-top: 1rem;

  div {
    display: flex;
    align-items: center;
  }
`

export const Spacer = styled.div`
  display: none;
  border-bottom: 1px solid var(--darkLiver);
  margin-top: 2rem;
  opacity: 0.4;

  ${mq('s')} {
    display: block;
  }
`
