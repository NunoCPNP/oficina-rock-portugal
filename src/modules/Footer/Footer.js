import { AiFillFacebook, AiFillInstagram, AiFillPhone, AiOutlineWhatsApp, AiTwotoneMail } from 'react-icons/ai'
import Image from 'next/image'
import Link from 'next/link'
import { SiTiktok } from 'react-icons/si'
import useTranslation from 'next-translate/useTranslation'

import Copyright from '@/components/Copyright'

import {
  ContactsContainer,
  GridContainer,
  GridItem,
  Item,
  MethodsContainer,
  Spacer,
  StyledFooter,
} from './Footer.styles'

const Footer = () => {
  const { t } = useTranslation('common')

  return (
    <StyledFooter data-testid="footer">
      <GridContainer>
        <GridItem>
          <Item>
            <Link href="/whoweare" passHref>
              <h4>{t(`who-we-are`)}</h4>
            </Link>
          </Item>
          <Item>
            <Link href="/privacypolicy" passHref>
              <h4>{t(`privacy-policies`)}</h4>
            </Link>
          </Item>
          <Item>
            <Link href="/returnpolicy" passHref>
              <h4>{t(`shipping-and-exchanges`)}</h4>
            </Link>
          </Item>
          {/* <Item>
            <Link href="/faq" passHref>
              <h4>{t(`faq`)}</h4>
            </Link>
          </Item> */}
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`contacts`)}</h3>
          <ContactsContainer>
            <div>
              <AiFillPhone data-testid="react-icon" />
              <AiOutlineWhatsApp data-testid="react-icon" />
              <span>+351 964 631 183</span>
            </div>
            <div>
              <AiFillFacebook data-testid="react-icon" />
              <AiFillInstagram data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
            <div>
              <AiTwotoneMail data-testid="react-icon" />
              <span>oficinarockpt@gmail.com</span>
            </div>
            <div>
              <SiTiktok data-testid="react-icon" />
              <span>oficinarockpt</span>
            </div>
          </ContactsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`payment-methods`)}</h3>
          <MethodsContainer>
            <div>
              <Image src="/payment_methods.png" alt={t(`payment-methods`)} width={160} height={35} />
            </div>
          </MethodsContainer>
          <Spacer />
        </GridItem>
        <GridItem>
          <h3>{t(`newsletter`)}</h3>
          <Spacer />
        </GridItem>
      </GridContainer>
      <Copyright />
    </StyledFooter>
  )
}

export default Footer
