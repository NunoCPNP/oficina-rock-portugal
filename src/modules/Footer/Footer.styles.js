import styled from '@emotion/styled'

export const StyledFooter = styled.footer`
  background-color: var(--ghostWhite);
`

export const GridContainer = styled.div`
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
`

export const GridItem = styled.div`
  h4 {
    font-weight: 400;
    line-height: 3rem;
  }
`
