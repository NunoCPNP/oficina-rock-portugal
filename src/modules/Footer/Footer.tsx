type Props = {}

const Footer: React.FC<Props> = () => {
  return <footer></footer>
}

export default Footer