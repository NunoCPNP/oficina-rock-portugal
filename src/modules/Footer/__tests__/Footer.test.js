import { render, cleanup } from '@testing-library/react'

import Footer from '../Footer'

afterEach(cleanup)

describe('<Footer /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<Footer />)

    const header = queryAllByTestId(/footer/i)
    expect(header).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<Footer />)

    expect(asFragment(<Footer />)).toMatchSnapshot()
  })
})