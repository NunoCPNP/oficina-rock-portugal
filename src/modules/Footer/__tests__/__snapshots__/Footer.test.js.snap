// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> spec Should match snapshot 1`] = `
.emotion-0 {
  background-color: var(--ghostWhite);
}

.emotion-1 {
  max-width: 140rem;
  margin: 0 auto;
  padding: 4rem;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
}

@media only screen and (max-width: 780px) {
  .emotion-1 {
    grid-template-columns: 1fr 1fr;
  }
}

@media only screen and (max-width: 480px) {
  .emotion-1 {
    grid-template-columns: 1fr;
  }
}

.emotion-2 {
  line-height: 3rem;
}

.emotion-2 h3 {
  text-transform: uppercase;
  padding-bottom: 1rem;
}

.emotion-2 h4 {
  font-weight: 400;
}

.emotion-3 {
  cursor: pointer;
}

.emotion-6 {
  display: none;
  border-bottom: 1px solid var(--darkLiver);
  margin-top: 2rem;
  opacity: 0.4;
}

@media only screen and (max-width: 480px) {
  .emotion-6 {
    display: block;
  }
}

.emotion-8 {
  padding-top: 1rem;
}

.emotion-8 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.4rem;
}

.emotion-8 div svg {
  width: 2.2rem;
  height: 2.2rem;
  margin-right: 0.6rem;
}

.emotion-11 {
  padding-top: 1rem;
}

.emotion-11 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 2rem;
}

<footer
  class="emotion-0"
  data-testid="footer"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <h4
          href="/whoweare"
        >
          common:who-we-are
        </h4>
      </div>
      <div
        class="emotion-3"
      >
        <h4
          href="/privacypolicy"
        >
          common:privacy-policies
        </h4>
      </div>
      <div
        class="emotion-3"
      >
        <h4
          href="/returnpolicy"
        >
          common:shipping-and-exchanges
        </h4>
      </div>
      <div
        class="emotion-6"
      />
    </div>
    <div
      class="emotion-2"
    >
      <h3>
        common:contacts
      </h3>
      <div
        class="emotion-8"
      >
        <div>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M885.6 230.2L779.1 123.8a80.83 80.83 0 0 0-57.3-23.8c-21.7 0-42.1 8.5-57.4 23.8L549.8 238.4a80.83 80.83 0 0 0-23.8 57.3c0 21.7 8.5 42.1 23.8 57.4l83.8 83.8A393.82 393.82 0 0 1 553.1 553 395.34 395.34 0 0 1 437 633.8L353.2 550a80.83 80.83 0 0 0-57.3-23.8c-21.7 0-42.1 8.5-57.4 23.8L123.8 664.5a80.89 80.89 0 0 0-23.8 57.4c0 21.7 8.5 42.1 23.8 57.4l106.3 106.3c24.4 24.5 58.1 38.4 92.7 38.4 7.3 0 14.3-.6 21.2-1.8 134.8-22.2 268.5-93.9 376.4-201.7C828.2 612.8 899.8 479.2 922.3 344c6.8-41.3-6.9-83.8-36.7-113.8z"
            />
          </svg>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            t="1569683925316"
            version="1.1"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs />
            <path
              d="M713.5 599.9c-10.9-5.6-65.2-32.2-75.3-35.8-10.1-3.8-17.5-5.6-24.8 5.6-7.4 11.1-28.4 35.8-35 43.3-6.4 7.4-12.9 8.3-23.8 2.8-64.8-32.4-107.3-57.8-150-131.1-11.3-19.5 11.3-18.1 32.4-60.2 3.6-7.4 1.8-13.7-1-19.3-2.8-5.6-24.8-59.8-34-81.9-8.9-21.5-18.1-18.5-24.8-18.9-6.4-0.4-13.7-0.4-21.1-0.4-7.4 0-19.3 2.8-29.4 13.7-10.1 11.1-38.6 37.8-38.6 92s39.5 106.7 44.9 114.1c5.6 7.4 77.7 118.6 188.4 166.5 70 30.2 97.4 32.8 132.4 27.6 21.3-3.2 65.2-26.6 74.3-52.5 9.1-25.8 9.1-47.9 6.4-52.5-2.7-4.9-10.1-7.7-21-13z"
            />
            <path
              d="M925.2 338.4c-22.6-53.7-55-101.9-96.3-143.3-41.3-41.3-89.5-73.8-143.3-96.3C630.6 75.7 572.2 64 512 64h-2c-60.6 0.3-119.3 12.3-174.5 35.9-53.3 22.8-101.1 55.2-142 96.5-40.9 41.3-73 89.3-95.2 142.8-23 55.4-34.6 114.3-34.3 174.9 0.3 69.4 16.9 138.3 48 199.9v152c0 25.4 20.6 46 46 46h152.1c61.6 31.1 130.5 47.7 199.9 48h2.1c59.9 0 118-11.6 172.7-34.3 53.5-22.3 101.6-54.3 142.8-95.2 41.3-40.9 73.8-88.7 96.5-142 23.6-55.2 35.6-113.9 35.9-174.5 0.3-60.9-11.5-120-34.8-175.6z m-151.1 438C704 845.8 611 884 512 884h-1.7c-60.3-0.3-120.2-15.3-173.1-43.5l-8.4-4.5H188V695.2l-4.5-8.4C155.3 633.9 140.3 574 140 513.7c-0.4-99.7 37.7-193.3 107.6-263.8 69.8-70.5 163.1-109.5 262.8-109.9h1.7c50 0 98.5 9.7 144.2 28.9 44.6 18.7 84.6 45.6 119 80 34.3 34.3 61.3 74.4 80 119 19.4 46.2 29.1 95.2 28.9 145.8-0.6 99.6-39.7 192.9-110.1 262.7z"
            />
          </svg>
          <span>
            +351 964 631 183
          </span>
        </div>
        <div>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
            />
          </svg>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 0 1-47.9 47.9z"
            />
          </svg>
          <span>
            oficinarockpt
          </span>
        </div>
        <div>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M477.5 536.3L135.9 270.7l-27.5-21.4 27.6 21.5V792h752V270.8L546.2 536.3a55.99 55.99 0 0 1-68.7 0z"
            />
            <path
              d="M876.3 198.8l39.3 50.5-27.6 21.5 27.7-21.5-39.3-50.5z"
            />
            <path
              d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-94.5 72.1L512 482 190.5 232.1h643zm54.5 38.7V792H136V270.8l-27.6-21.5 27.5 21.4 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5h.1l39.3 50.5-27.7 21.5z"
            />
          </svg>
          <span>
            oficinarockpt@gmail.com
          </span>
        </div>
        <div>
          <svg
            data-testid="react-icon"
            fill="currentColor"
            height="1em"
            role="img"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title />
            <path
              d="M12.53.02C13.84 0 15.14.01 16.44 0c.08 1.53.63 3.09 1.75 4.17 1.12 1.11 2.7 1.62 4.24 1.79v4.03c-1.44-.05-2.89-.35-4.2-.97-.57-.26-1.1-.59-1.62-.93-.01 2.92.01 5.84-.02 8.75-.08 1.4-.54 2.79-1.35 3.94-1.31 1.92-3.58 3.17-5.91 3.21-1.43.08-2.86-.31-4.08-1.03-2.02-1.19-3.44-3.37-3.65-5.71-.02-.5-.03-1-.01-1.49.18-1.9 1.12-3.72 2.58-4.96 1.66-1.44 3.98-2.13 6.15-1.72.02 1.48-.04 2.96-.04 4.44-.99-.32-2.15-.23-3.02.37-.63.41-1.11 1.04-1.36 1.75-.21.51-.15 1.07-.14 1.61.24 1.64 1.82 3.02 3.5 2.87 1.12-.01 2.19-.66 2.77-1.61.19-.33.4-.67.41-1.06.1-1.79.06-3.57.07-5.36.01-4.03-.01-8.05.02-12.07z"
            />
          </svg>
          <span>
            oficinarockpt
          </span>
        </div>
      </div>
      <div
        class="emotion-6"
      />
    </div>
    <div
      class="emotion-2"
    >
      <h3>
        common:payment-methods
      </h3>
      <div
        class="emotion-11"
      >
        <div>
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjM1IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="common:payment-methods"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </div>
        </div>
      </div>
      <div
        class="emotion-6"
      />
    </div>
    <div
      class="emotion-2"
    >
      <h3>
        common:newsletter
      </h3>
      <div
        class="emotion-6"
      />
    </div>
  </div>
  <div
    class="emotion-15"
    data-testid="copyright"
  >
    <div>
      common:copyright
    </div>
  </div>
</footer>
`;
